src/api/c++/z3++.h

```diff
@@ -385,6 +412,10 @@ namespace z3 {
         context m_ctx;
     public:
         scoped_context(Z3_context c): m_ctx(c) {}
+        scoped_context(scoped_context const &) = delete;
```

Comment:
These are not needed. context already forbids copying. And the move constructors aren't useful. No one will move this thing.

Reply:
I have the copy functions in just because of the rule of 5, even if we happen to be doing what it would be implicit in the first place. As for move semantics, this is an API, I see no reason to disallow a feature that has almost no downsides to permitting.

Reply:
this is an internal class. there's no need for this stuff

Reply:
I see. If we don't care about extensibility of the superclass I can delete them.